webcam-capture-drivers/driver-ipcam/src/main/java/com/github/sarxos/webcam/ds/ipcam/IpCamDevice.java

```diff
@@ -235,30 +86,71 @@ public IpCamDevice(String name, URL url, IpCamMode mode, IpCamAuth auth) {
 		}
 	}
 
-	public IpCamHttpClient getClient() {
-		return client;
-	}
-
-	protected static final URL toURL(String url) {
-
-		String base = null;
-		if (url.startsWith("http://")) {
-			base = url;
-		} else {
-			base = String.format("http://%s", url);
-		}
-
-		try {
-			return new URL(base);
-		} catch (MalformedURLException e) {
-			throw new WebcamException(String.format("Incorrect URL '%s'", url), e);
-		}
-	}
-
-	@Override
-	public String getName() {
-		return name;
-	}
+    protected static final URL toURL(String url) {
+
+        String base = null;
+        if (url.startsWith("http://")) {
+            base = url;
+        } else {
+            base = String.format("http://%s", url);
+        }
+
+        try {
+            return new URL(base);
+        } catch (MalformedURLException e) {
+            throw new WebcamException(String.format("Incorrect URL '%s'", url), e);
+        }
+    }
+
+    private void initThreadPool() {
+        queue = new ArrayBlockingQueue<Runnable>(1);
+        executorService =
+                new ThreadPoolExecutor(

```

Comment:
Hi @bocse, can we make ```IpCamDevice``` implements ```ThreadFactory``` so we can use ```this``` as the argument here to avoid these nested anonymous classes? In this case we can do:

```java
= new ThreadPoolExecutor(1, 1, 1, TimeUnit.SECONDS, queue, this);
```